classes/Course.php

```diff
@@ -2265,16 +2266,15 @@ public function attach_product_with_course( $post_ID, $post_data ) {
 
 		if ( 'wc' === $monetize_by ) {
 
-			$is_update = ( $attached_product_id && wc_get_product( $attached_product_id ) ) ? true : false;
+			$is_update = ( $product_id && wc_get_product( $product_id ) ) ? true : false;
```

Comment:
Here we are calling `wc_get_product` fn and on the line no: **2275** again calling `wc_get_product`
Which results in unoptimized code. Instead, we can keep it in a variable, then use it where required.

Reply:
We can't do that because the price needs to be kept up to date during creation or updates.